src/storage/src/hummock/sstable/block_iterator.rs

```diff
@@ -92,18 +93,20 @@ impl BlockIterator {
         self.next_until_prev_offset(self.block.len());
     }
 
-    pub fn seek(&mut self, key: &[u8]) {
-        self.seek_restart_point_by_key(key);
-        self.next_until_key(key);
+    pub fn seek(&mut self, key: FullKey<&[u8]>) {
+        let full_key_encoded = key.encode();
```

Comment:
I think we can reuse `self.key` as a buffer to hold the encoded key to avoid an extra memory allocation here.

Reply:
This interface, along with `seel_le` will be changed in next pr, which will encode without table_id.